visualization/app/codeCharta/ui/fileChooser/fileChooser.po.ts

```diff
@@ -19,4 +19,13 @@ export class FileChooserPageObject {
 
 		await fileChooser.cancel()
 	}
+
+	public static async selectFile(path: string) {
+		const fileChooser = await page.waitForFileChooser()
+
+		await fileChooser.accept([path])
+
+		await page.waitForSelector("#loading-gif-file")
+		await page.waitForSelector("#loading-gif-file", { visible: false })
+	}
```

Comment:
What about unifying `openFile` and `selectFile`? They share most code and you could just use an option to define if the click should be checked as well as in:

```js
openFile({ path, clickOnFileChooser = true }) {
  const [fileChooser] = await Promise.all([
    page.waitForFileChooser(),
    clickOnFileChooser && expect(page).toClick(...)
  ])
  await fileChooser.accept([path])
  ...
}
```

Reply:
@JaiJoPa If you're not a fan of `clickOnFileChooser && expect(page).toClick(...)`, you can do something like this as well:

```
public async openFiles(paths: string[], toOpenFileChooser = true) {
	function openFileChooser(toOpenFileChooser: boolean) {
		if(toOpenFileChooser) {
			return expect(page).toClick("file-chooser-directive .toolbar-button", { timeout: 3000 })
		}
	}

	const [fileChooser] = await Promise.all([
		page.waitForFileChooser(),
		openFileChooser(toOpenFileChooser)
	])

	await fileChooser.accept(paths)

	await page.waitForSelector("#loading-gif-file")
	await page.waitForSelector("#loading-gif-file", { visible: false })
}
```

@BridgeAR suggestion is shorter though, and in that case I like the shorter syntax more.
Usually I'm not really a fan of it yet. ^^'